### PR TITLE
updates code guidelines, creates linting file

### PR DESCRIPTION
**Pull request by matthewcarleton**
This removes all of the syntax that can be run through a linter to make the guidelines a bit more clear.

### CODE REVIEW

File: CODE-GUIDELINES.md [hunk 1]
```diff
@@ -128,23 +128,7 @@ For example:
...
-- End all declarations with a semi-colon, even the last one.
-- Comma-separated property values should include a space after each comma (e.g., `box-shadow`).
-- Don't include spaces after commas within `rgb()`, `rgba()`, `hsl()`, `hsla()`, or `rect()` values. This helps differentiate multiple color values (comma, no space) from multiple property values (comma with space).
-- Lowercase all hex values, e.g., `#fff`. Lowercase letters are much easier to discern when scanning a document as they tend to have more unique shapes.
-- Use shorthand hex values where available, e.g., `#fff` instead of `#ffffff`.
-- Quote attribute values in selectors, e.g., `input[type="text"]`.  They’re only optional in some cases, and it’s a good practice for consistency.
-- Don't prefix property values or color parameters with a leading zero (e.g., `.5` instead of `0.5` and `-.5em` instead of `-0.5em`).
-- Avoid specifying units for zero values, e.g., `margin: 0;` instead of `margin: 0px;`.
-
+Patternfly uses [stylelint](https://stylelint.io) to enforce the following formatting:
```
Comment (srambach) [inline CODE-GUIDELINES.md]: Link here to the stylelint json for details?

Comment (srambach) [reply CODE-GUIDELINES.md]: Since you have all the rules in comments now, I think it would be nice, so that those rules are still documented.

Comment (matthewcarleton) [reply CODE-GUIDELINES.md]: ok that makes sense, Thanks Sarah :)

**Comment by andresgalante** [comment]
@matthewcarleton What is package-lock.json doing here? can we remove it for this PR, and maybe have a new PR with updates for patternlab

**Pull request merged**